app/sign.js

```diff
@@ -374,27 +382,32 @@ const handleSign = function(args) {
 
   let txHex, halfSignedInfo;
 
-  switch (coin) {
-    case 'eth': case 'teth':
+  let basecoin;
+  try {
```

Comment:
I'm not a big fan of try-catch being used for flow control - can we change this to an `if (recoveryRequest.tokenContractAddress)...`?

Reply:
ok, can change